### ISSUE TITLE
(Bug) Could not get duration (also suggestion)

### ISSUE THREAD

**Issue by skil3z** [TeamNewPipe/NewPipeExtractor#110]
## Exception
* __User Action:__ requested stream
* __Request:__ https://www.youtube.com/watch?v=UwLtyvGdNbc
* __Content Language:__ FI
* __Service:__ YouTube
* __Version:__ 0.14.1
* __OS:__ Linux Android 4.4.4 - 19

(This happens with EVERY video btw)

The problem is that I live in Finland and we have _some_ inconsistencies in following international standards, especially in time/decimal notation, and here the time separator is . (dot) instead of the more common : (punctuation colon).

The problem is that when this program expects a : in the timeformat thing, but youtube returns instead a dot (based on geographic location), which causes errors and in some cases for the app to hang partially.
For example 4:48 is 4.48 here (it also looks awful, like a decimal number, national standards cannot be done anything to though :) gotta just live with it)

I do not have the technical knowhow to compile this program again (well I managed to compile this extractor part to .jar but don't know how to shove it into the main program / compile it), BUT I know something about programming and I determined that the problem is fixed by changing this on line 33 in the file 

/NewPipeExtractor/extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeParsingHelper.java

from 
```
String[] splitInput = input.split(":");
```
to
```
String[] splitInput = input.split("\\."); 
```
Could the fix be to add something before that split thing that the split character equals : if content language variable equals something? Like (don't know exactly but) 
```
if contentLang = "US" then splitCharacter = ":" 
if contentLang = "FI" then splitCharacter = "\\."
```
 (of course it would be required to look at the wikipedia page for what countries actually have dot and which do not.. **and to write that line in proper java syntax lol** I can help with the _former_ one if needed.

Sorry if there's something wrong with this. Thank you. This program is incredible btw, I use it everyday

Here's the long error:
```
org.schabi.newpipe.extractor.exceptions.ParsingException: Could not get Duration: https://www.youtube.com/watch?v=S7de4JJVV8A
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamInfoItemExtractor.getDuration(YoutubeStreamInfoItemExtractor.java:93)
	at org.schabi.newpipe.extractor.stream.StreamInfoItemsCollector.extract(StreamInfoItemsCollector.java:54)
	at org.schabi.newpipe.extractor.stream.StreamInfoItemsCollector.commit(StreamInfoItemsCollector.java:89)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.getRelatedVideos(YoutubeStreamExtractor.java:517)
	at org.schabi.newpipe.extractor.utils.ExtractorHelper.getRelatedVideosOrLogError(ExtractorHelper.java:32)
	at org.schabi.newpipe.extractor.stream.StreamInfo.extractOptionalData(StreamInfo.java:249)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:64)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:55)
	at org.schabi.newpipe.util.ExtractorHelper.lambda$getStreamInfo$3$ExtractorHelper(ExtractorHelper.java:113)
	at org.schabi.newpipe.util.ExtractorHelper$$Lambda$3.call(Unknown Source)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3310)
	at io.reactivex.internal.operators.single.SingleDoOnSuccess.subscribeActual(SingleDoOnSuccess.java:35)
	at io.reactivex.Single.subscribe(Single.java:3310)
	at io.reactivex.internal.operators.maybe.MaybeFromSingle.subscribeActual(MaybeFromSingle.java:41)
	at io.reactivex.Maybe.subscribe(Maybe.java:4073)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.drain(MaybeConcatArray.java:153)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.request(MaybeConcatArray.java:78)
	at io.reactivex.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onSubscribe(FlowableElementAtMaybe.java:66)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray.subscribeActual(MaybeConcatArray.java:42)
	at io.reactivex.Flowable.subscribe(Flowable.java:14349)
	at io.reactivex.internal.operators.flowable.FlowableElementAtMaybe.subscribeActual(FlowableElementAtMaybe.java:36)
	at io.reactivex.Maybe.subscribe(Maybe.java:4073)
	at io.reactivex.internal.operators.maybe.MaybeToSingle.subscribeActual(MaybeToSingle.java:46)
	at io.reactivex.Single.subscribe(Single.java:3310)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:579)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:841)
Caused by: java.lang.NumberFormatException: Invalid long: "5.42"
	at java.lang.Long.invalidLong(Long.java:124)
	at java.lang.Long.parse(Long.java:361)
	at java.lang.Long.parseLong(Long.java:352)
	at java.lang.Long.parseLong(Long.java:318)
	at org.schabi.newpipe.extractor.services.youtube.linkHandler.YoutubeParsingHelper.parseDurationString(YoutubeParsingHelper.java:64)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamInfoItemExtractor.getDuration(YoutubeStreamInfoItemExtractor.java:91)
	... 34 more

-------------------

(the same as above, repeated 10 times for all the suggested videos)

-------------------

```

**Comment by theScrabi** [comment]
@skil3z nice work :)
Maybe https://teamnewpipe.github.io/dcumentation might help you. If you manage to send a Pullrequest I will review it :)

**Comment by theScrabi** [comment]
This is still missing, because I was trying to focuse on th extractor. But you are right we need a section to tell how to use jitpack.io.

In short: In order to test your changes with the frintend you need to upload them to sour own github repository, and then post a link to it at jitpack.io . Now take the latest commit that is displayed there and past it as the line you pointed out.

**Comment by theScrabi** [comment]
... I will write a better description than this :)

**Comment by skil3z** [comment]
Ha! I managed to cheat this. I left gradle daemon running in the background, so it wouldn't download the files, but use the already cached ones in /home/myuser/.gradle/caches if I recompile NewPipe, BUT I overwrote those cached files with the ones I got from compiling my edit of NewPipeExtractor, and it actually worked :DD No errors or anything, apk runs on my real phone and the bug is fixed :) 

I will tomorrow try to make something more sensible out of this. Pullrequest maybe? This is my first day doing anything with github or java :)
